File: Code Optimization and Debugging II/Python/BubbleSort.py
def bubble_sort(arr):
    size = len(arr)
    
    for i in range(0, size-1):
        for j in range(0, size - 1 - i):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

arr = [int(x) for x in input().split()]
bubble_sort(arr)
print(*arr)

# Ip: 20 50 40 30 10
# Op: 10 20 30 40 50

File: Code Optimization and Debugging II/C/BubbleSort.c
#include <stdio.h>

void sort(int arr[], int size) {
    for (int i = 0; i < size - 1; i++) {
        for (int j = 0; j < size - 1 - i; j++) {
            if (arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

int main() {
    int size = 10;
    int arr[size];

    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }

    sort(arr, size);

    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
}

/*

Input: 10 9 8 7 6 5 4 3 2 1
Output: 1 2 3 4 5 6 7 8 9 10

*/